python/servo/bootstrapper/base.py

```diff
@@ -0,0 +1,78 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from __future__ import print_function, unicode_literals
+
+import os
+import subprocess
+
+from servo.command_base import CommandBase
+
+
+class BaseBootstrapper(CommandBase):
+    """Base class for system bootstrappers."""
+
+    def __init__(self, interactive=False):
+        self.package_manager_updated = False
+        self.interactive = interactive
+
+    def install_packages(self):
+        '''
+        Install packages required to build Servo.
+        '''
+        raise NotImplementedError('%s must implement install_system_packages()' %
+                                  __name__)
+
+    def install_mobile_android_packages(self):
+        '''
+        Install packages required to build Servo for Android.
+        '''
+        raise NotImplementedError('Cannot bootstrap Servo for Android: '
+                                  '%s does not yet implement install_mobile_android_packages()'
+                                  % __name__)
+
+    def which(self, name):
+        """Python implementation of which.
```

Comment:
Can't we just use `distutils.spawn.find_executable`? (like we do in `mach_bootstrap.py`)